03-Bash/find_dup_files/find_dup_files.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash +x
+
+# set default dir
+sourceDir=$(pwd)
+
+# set new source dir if available
+if [ $# -ne 0 ]; then 
+    sourceDir=$1
+fi
+
+cd $sourceDir
+# create list of files in dir
+find . * -maxdepth 0 -type f > fileList
+
+# check if file list created
+if [ ! -e fileList ]; then
+    echo cant create fileList
+    exit 1
+fi
+
+#create file with checksum for each file
+cat fileList | while read line
+do
+    tmp=$(md5sum "$line") 
+    echo $tmp | cut -d ' ' -f1 > "$line".md5sum
+done
+
+# compare contents of *.md5sum files
+cat fileList | while read md5File1
```

Comment:
The complexity of this algorithm is O^2. Please consider another way for finding duplicate files.

Reply:
Do you mean the md5sum comparison mechanism?

Reply:
Yes. You compare each of N records with other. But it is possible to sort all records and compare only the neighbors.